File: beginner/__1000__/1045/1045.php
<?php 

    /**
     * FUNCIONA MAS O SITE ME BLOQUEOU! KK
     */

    function bubbleSort($lista){
        $len = count($lista);
        for($i = 0; $i < $len; $i++){
            for($j = 0; $j < $len - $i - 1; $j++){
                if($lista[$j] > $lista[$j + 1]){
                    $temp = $lista[$j + 1];
                    $lista[$j + 1] = $lista[$j];
                    $lista[$j] = $temp;
                }
            }
        }
        return $lista;
    }

    $INPUT = explode(" ", readline());
    $data = [];
    for($i = 0; $i < 3; $i++){
        $data[$i] = (float) $input[$i];
    }

    list($C, $B, $A) = bubbleSort($data);

    $a2 = $A * $A;
    $bc2 = ($B * $B) + ($C * $C);

    if($A >= $B + $C) echo "NAO FORMA TRIANGULO\n";
    else {
        if(abs($a2 - $bc2) < 0.000000001 ) echo "TRIANGULO RETANGULO\n";
        if($a2 > $bc2) echo "TRIANGULO OBTUSANGULO\n";
        if($a2 <  $bc2) echo "TRIANGULO ACUTANGULO\n";
        if($A == $B && $B == $C) echo "TRIANGULO EQUILATERO\n";
        else if($A == $B || $A == $C || $C == $B){
            echo "TRIANGULO ISOSCELES\n";
        }
    }
?>

File: beginner/__1000__/1045/1045.java
import java.util.Scanner;
import java.util.Arrays;

public class Main {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        String[] data = in.nextLine().split(" ");
        double[] dataConv = new double[3];

        for(int i = 0; i < 3; i++)
            dataConv[i] = Double.parseDouble(data[i]);

        Arrays.sort(dataConv);
        double A = dataConv[2];
        double B = dataConv[1];
        double C = dataConv[0];

        double a2 = A * A;
        double bc2 = (B * B) + (C * C);

        if(A >= B + C){ System.out.println("NAO FORMA TRIANGULO");} 
        else {
            if(Math.abs(a2 - bc2) < 1e-9 ){ System.out.println("TRIANGULO RETANGULO");}
            if(a2 > bc2){ System.out.println("TRIANGULO OBTUSANGULO");}
            if(a2 <  bc2){ System.out.println("TRIANGULO ACUTANGULO");}
            if(A == B && B == C){ System.out.println("TRIANGULO EQUILATERO");}
            else if(A == B || A == C || C == B){
                System.out.println("TRIANGULO ISOSCELES");
            }
        }

        in.close();
    }
}